### PR TITLE
WIP: lmdb support

### PR DESCRIPTION
**Pull request by uhliarik**
There is still WIP.

### CODE REVIEW

**Comment by minfrin** [comment]
Not had a chance to review, but definite +1 in principle.

**Comment by uhliarik** [comment]
Since I had to rebase this code to the trunk, I'm no longer able to run tests. If anyone can see where I'm missing linking the library, feel free to review the commit. 

```
$ make test
cd test && make all check
make[1]: Entering directory '/home/luhliari/playground/UPSTREAM/APR/apr_lmdb/test'
make[2]: Entering directory '/home/luhliari/playground/UPSTREAM/APR/apr_lmdb/test'
/bin/sh /home/luhliari/playground/UPSTREAM/APR/apr_lmdb/libtool --silent --mode=compile --tag=CC gcc -g -O2 -pthread   -DHAVE_CONFIG_H  -DLINUX -D_REENTRANT -D_GNU_SOURCE   -I../include -I./../include  -o globalmutexchild.lo -c globalmutexchild.c && touch globalmutexchild.lo
/bin/sh /home/luhliari/playground/UPSTREAM/APR/apr_lmdb/libtool --silent --mode=link --tag=CC gcc -g -O2 -pthread   -DHAVE_CONFIG_H  -DLINUX -D_REENTRANT -D_GNU_SOURCE   -I../include -I./../include   -no-install    -o globalmutexchild globalmutexchild.lo ../libapr-2.la   -lcrypt -lrt -lcrypt  -lpthread -ldl -lexpat
/usr/bin/ld: ../.libs/libapr-2.so: undefined reference to `mdb_drop'
/usr/bin/ld: ../.libs/libapr-2.so: undefined reference to `mdb_env_close'
/usr/bin/ld: ../.libs/libapr-2.so: undefined reference to `mdb_cursor_get'
/usr/bin/ld: ../.libs/libapr-2.so: undefined reference to `mdb_del'
/usr/bin/ld: ../.libs/libapr-2.so: undefined reference to `mdb_env_open'
/usr/bin/ld: ../.libs/libapr-2.so: undefined reference to `mdb_strerror'
/usr/bin/ld: ../.libs/libapr-2.so: undefined reference to `mdb_cursor_open'
/usr/bin/ld: ../.libs/libapr-2.so: undefined reference to `mdb_txn_commit'
/usr/bin/ld: ../.libs/libapr-2.so: undefined reference to `mdb_dbi_close'
/usr/bin/ld: ../.libs/libapr-2.so: undefined reference to `mdb_dbi_open'
/usr/bin/ld: ../.libs/libapr-2.so: undefined reference to `mdb_cursor_close'
/usr/bin/ld: ../.libs/libapr-2.so: undefined reference to `mdb_env_create'
/usr/bin/ld: ../.libs/libapr-2.so: undefined reference to `mdb_txn_begin'
/usr/bin/ld: ../.libs/libapr-2.so: undefined reference to `mdb_env_set_mapsize'
/usr/bin/ld: ../.libs/libapr-2.so: undefined reference to `mdb_put'
/usr/bin/ld: ../.libs/libapr-2.so: undefined reference to `mdb_get'
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:125: globalmutexchild] Error 1
make[2]: Leaving directory '/home/luhliari/playground/UPSTREAM/APR/apr_lmdb/test'
make[1]: *** [/home/luhliari/playground/UPSTREAM/APR/apr_lmdb/build/apr_rules.mk:118: all-recursive] Error 1
make[1]: Leaving directory '/home/luhliari/playground/UPSTREAM/APR/apr_lmdb/test'
```

File: build.conf [hunk 1]
```diff
@@ -22,6 +22,7 @@ paths =
   crypto/crypt_blowfish.c
   dbm/apr_dbm_sdbm.c
   dbm/apr_dbm.c
+  dbm/apr_dbm_lmdb.c
```
Comment (notroj) [inline build.conf]: This should be removed, the DBM drivers which can be built as DSOs should not be listed here otherwise it will get linked twice.

**Comment by notroj** [comment]
With the line marked in my review above removed, and e147886e92b2858895fe9fff30fccf9dba678cad reverted, I can build with:

```sh
"./configure" \
"--with-lmdb" \
"--with-berkeley-db" \
"--with-dbm=db5" \
"--disable-modular-dso"
```

and this passes the tests. Thanks @uhliarik!

**Comment by notroj** [comment]
I've tested this with mod_dav and it seems to work well. As well as the changes described above I changed the `_usednames` implementation to pass back the "-lock" file as well:

https://github.com/notroj/apr/commit/26afe6b5b134ef3652bb629e6c68af38317f7f62

**Comment by uhliarik** [comment]
Thanks Joe for reviewing my changes. Previously, I was configurating the lmdb driver with:

```
./configure --with-lmdb --with-dbm=lmdb
```

And I was having troubles with loading lmdb driver as DSO. It was failing during DSO loading phase. I hope there is no issue with this and it is just some configuration thing.

**Comment by notroj** [comment]
Thanks a lot for the contribution, @uhliarik

**Pull request closed without merging**